api/client/run.go

```diff
@@ -36,6 +38,27 @@ func (cid *cidFile) Write(id string) error {
 	return nil
 }
 
+// Exit code 125 signals docker failure
+// Exit code 126 signals contained cmd found but cannot be invoked
+// Exit code 127 for contained cmd cannot be found
+func runError(err error) error {
+	// syscall.ENOENT.Error() and exec.ErrNotFound.Error() strings vary per client
+	// Windows: "The system cannot find the file specified" and "executable file not found in %PATH%"
+	// Linux: "no such file or directory" and "executable file not found in $PATH"
+	if strings.Contains(err.Error(), "system cannot find the file specified") ||
+		strings.Contains(err.Error(), "no such file or directory") ||
+		strings.Contains(err.Error(), "executable file not found") {
+		return Cli.StatusError{StatusCode: 127}
+	}
+	if strings.Contains(err.Error(), syscall.EACCES.Error()) {
+		return Cli.StatusError{StatusCode: 126}
+	}
+	if strings.Contains(err.Error(), "latest not found") {
```

Comment:
```
# docker run busybox:notlatest true
Unable to find image 'busybox:notlatest' locally
Pulling repository docker.io/library/busybox
Tag notlatest not found in repository docker.io/library/busybox

# echo $?
1

# docker run cpuguy83/runcomtest          
Unable to find image 'cpuguy83/runcomtest:latest' locally
Pulling repository docker.io/cpuguy83/runcomtest

# echo $?
125
```
If I'm trying to pull an image by a not-latest tag the error is shown but the exit code is `1`
If I'm trying to pull an image looking for latest tag but it doesn't have it error isn't shown and exit code is `127`
instead of looking for `latest not found` should this look for `not found in repository` to be consistent for every tag? (not sure why you don't want to show the error if the tag isn't latest)